### ISSUE TITLE
Add emojiOne support

### ISSUE THREAD

**Issue by Bubu** [slp/Telegram-FOSS#53]
Initial request was in #35.

This is important for anything < Android 4.4 as there are either no emoji or only b/w ones.

Wip here: #50
Todo:
- Option to switch emojis
- adjust to new emoji code
- testing

**Comment by Ecron** [comment]
I wish I could help more actively, but I'm just a mere user with no coding knowledge.

Still I offer my help for testing purposes, at least I can do that :+1:

**Comment by Ecron** [comment]
Now EmojiOnes sets the release of the 2016 Collection to December 17th (you can check their Instagram's description).

Just for everybody to know :+1:

**Comment by Turakar** [comment]
+1

**Comment by emunicio** [comment]
Yes! I only have the android native emoji's (don't have the ones from google) and not only they are ugly but also some of them are missing. So it would nice fix this problem
regards

**Comment by Turakar** [comment]
The emojis from android and emojiOne are also not always equal.

**Comment by mcagl** [comment]
I'm yet on Telegram-FOSS 3.1.3 and I'm not upgrading it only because of the *ugly* and *incomplete* black android stock emojis (I cannot tell one emoji from another) (unfortunately I have android 4.1.1 and cannot easily upgrade it without buying another phone), so +1 for integrated EmojiOne in Telegram-FOSS :)

**Comment by Turakar** [comment]
For compatibilty reasons it would be good to have the same emojis as in the normal Telegram app. However I don't know the license.

**Comment by mcagl** [comment]
@Turakar Yeah, I know, but there were months and months of discussions on that: #35 
TL;DR: Those emojis are Apple's.

**Comment by kq01526** [comment]
The new Emoji One 2.0 has been released today:

http://emojione.com/
https://github.com/Ranks/emojione

:wink:

**Comment by jomo** [comment]
This is a good idea, but honestly I'd prefer if all my apps would just stick to the system emoji instead of bringing their own. _(Even though I'm probably going to use EmojiOne as my system emoji anyway)_

**Comment by Ecron** [comment]
@jomo It's supposed to be an option to switch off EmojiOne set and use system's.

**Comment by emunicio** [comment]
Any idea when the problem with the emojis will be fixed? I have just updated to 3.4.2 but the problem is still there: I can only see black and white ugly emojis and some of them are missing. Is this normal?

**Comment by Bubu** [comment]
No one had time to implement this yet, sorry.

**Comment by Bubu** [comment]
In the new version there will be missing emoji for everyone expect Android 6.0.1 users.
But on the bright side it will now be easier to integrate a different emoji set. I'm quite busy right now but I'll try to look at this again when I have some time.

**Comment by Peque** [comment]
@Bubu: thanks for your hard work. :blush:

**Comment by Cotopaxi42** [comment]
Thanks!
Are the missing emojis new ones or old ones that we can't have anymore?

**Comment by Cotopaxi42** [comment]
Sorry for double-posting. 
Issue on my Phone (Adroid 5.1.1): I can see the emojis in the "home" menu (where you see all your chats) but I can't see them in the chats themselves, they don't appear...

**Comment by NeroBurner** [comment]
Same on 4.4.4

**Comment by NeroBurner** [comment]
Emojis (android ones) are here again with 3.6.1a

**Comment by Allanitomwesh** [comment]
Hello,I have an issue with the emojis,where some are Android ones,and some are EmojiOne ones,and then there are gaps that are blank
Is there a way to use the full set of EmojiOne emojis? I don't like the Android jelly bean emojis and EmojiOne has no gaps

**Comment by Ecron** [comment]
@Allanitomwesh As far as I know, the EmojiOne set is not implemented yet here (www.emojione.com). How can you have issues with that iconset? Could you explain better, please?

**Comment by Allanitomwesh** [comment]
Some emojis are simply a rectangle with an X in it,when am scrolling emojis,or when someone else posts one in a chat. I'm assuming these are emojis apple have but stock android does not? (My android version is 5.0.2) I see there's a new update, maybe it fixes iit? I'll update and observe

**Comment by kq01526** [comment]
> @emunicio wrote:
> I only have the android native emoji's (don't have the ones from google) and not only they are ugly but also some of them are missing

It might make sense if you guys let Google know about your frustration with the Android (Noto Color Emoji) Emojis over there:

https://github.com/googlei18n/noto-emoji/issues/9

:wink:

**Comment by Allanitomwesh** [comment]
Okay after the update still the same,also ive noticed I was simply confusing some of the emoji as EmojiOne since they look similar. There are gaps on stock android emoji (rectangles with an x)

**Comment by Bubu** [comment]
@Allanitomwesh Yes, only Android 6.0.1+ supports (almost) all of Telegram's emojis. Unfortunately there is no easy to use list to use which ones are not supported by Android 4.4/5.x. Otherwise we could filter these on older systems.
But I think I will spend time I find for working on this toward integrating the EmojiOne set.

**Comment by PanderMusubi** [comment]
http://m.slashdot.org/story/309723

**Comment by Allanitomwesh** [comment]
@bubu I would like to help with any testing. I dunno how to program though.

**Comment by arielenter** [comment]
Installed version 3.10.1 from f-droit. EmojiOne haven't been implemented yet. I'll look to make a donation to the project. I hope this feature can get implemented soon :)

Thanks a lot for the great work C:

**Comment by Ecron** [comment]
This project seems a little in stand by, that's true.

@slp @Bubu Do you need any help here? Any update on the status of the project, maybe?

Thanks ;-)

**Comment by thermatk** [comment]
Hi!
I've taken a different approach, so I generated the twemoji into the asset files with Telegram's structure. EmojiOne can't be an option until they update to include Emoji 4.0 standard: bug is here https://github.com/Ranks/emojione/issues/382.
One can just replace the files in TMessagesProj/src/main/assets/emoji/
with those i have here: https://github.com/thermatk/TgEmojiReplacer/tree/master/EMOJITOOLS/imgs/tinified . Should I send you a pull request with them @slp?
Everything to generate these pics from latest https://github.com/twitter/twemoji/releases/tag/v2.2.1 is in the folder, but it's very much a hacky workaround based on the workaround the whole emoji subsystem in this app.

As for the option in the settings - with plain image replacements it will be pretty easy to maintain, just add the option to the settings area and replace the absolute path to assets with one including a preference. Yet putting multiple image libraries will grow the apk size a bit :) The only problem is that there are only two fonts currently ready for 4.0 - twemoji which I generated and Google Noto.
What do you think?

**Comment by thermatk** [comment]
Google Noto replacements are ready as well:
https://github.com/thermatk/TgEmojiReplacer/tree/master/EMOJITOOLS/GoogleNoto/tinified

**Comment by thermatk** [comment]
generate emojiOne as well, falling back to twemoji in case there is no emoji yet :)
Moved everything to a separate repo:
https://github.com/thermatk/TelegramEmojiGenerator

**Comment by Allanitomwesh** [comment]
Sweet, @Bubu @slp should totally check this out for the enxt F-Droid update

**Comment by slp** [comment]
@thermatk That's cool, thanks. I've just replaced the conflicting assets with the EmojiOne+Twemoji version, and restored the original Emoji code.

**Comment by Peque** [comment]
The new emoji set is beautiful and works really well. Thanks for this change! :heart:

**Comment by Bubu** [comment]
Thanks @thermatk and @slp! Closing.

**Comment by c33s** [comment]
as a user who hasn't upgraded yet i have a question:
is the new emoji set optional or am i forced to use it? really want to keep my system emojis.

**Comment by emunicio** [comment]
Big thanks for the new emoji set! Now really looks nice! :-D

**Comment by Bubu** [comment]
It is, unfortunately, not optional. I also prefer the android system emoji, but it takes much more time than we currently have for this project to develop (and more importantly maintain) the option to switch the new emojis on and off.
If someone can do that we could probably merge that, but until that happens the emojione set it is.

**Comment by thermatk** [comment]
@c33s If you have Xposed, you can use my module(made for PlayStore version, but I think it will work) and switch to Google Noto:
[TelegramReEmoji](http://repo.xposed.info/module/com.thermatk.android.xf.telegram)
:)

**Comment by Allanitomwesh** [comment]
Love the new emojiset great work.
